### ISSUE TITLE
Handle time rollover

### ISSUE THREAD

**Issue by wetheredge** [blackbox-log/blackbox-log#54]
System time gets logged in microseconds as an unsigned 32-bit integer[^1]. This gives a max uptime of just over 1 hour 11 minutes before it rolls over. `blackbox_decode` [handles this](https://github.com/betaflight/blackbox-tools/blob/439a9930672877a0d9667d1a5a591549c6a48392/src/parser.c#L1060-L1081) with a heuristic that tries to differentiate rollover from skips. Neither Betaflight's nor INAV's log viewer seems to handle rollover.

See https://github.com/iNavFlight/blackbox-log-viewer/issues/51. Unfortunately, the log there is from INAV 2.6.0 which is not planned to be supported, so a new example would be ideal.

[^1]: See the [documentation](https://betaflight.com/docs/development/Blackbox%20Internals#main-frames-i-p) and [Betaflight's](https://github.com/betaflight/betaflight/blob/4.4.0/src/main/blackbox/blackbox.c#L185) and [INAV's](https://github.com/iNavFlight/inav/blob/5.1.0/src/main/blackbox/blackbox.c#L206) blackbox definitions

**Comment by lwan288** [comment]
Using blackbox_decode within MultiWii Planner can fix avoid issue? 

> 

Using blackbox_decode within MultiWii Planner can avoid this issue?

**Comment by wetheredge** [comment]
@lwan288 Yes, it seems that [flightlog2x](https://github.com/stronnag/bbl2kml) that MultiWii Planner uses to parse blackbox logs [shells out to `blackbox_decode`](https://github.com/stronnag/bbl2kml/blob/43b7254539516ddb3e136fb7d18bee16a6a906cc/pkg/bbl/bblreader.go#L670-L672), so it should handle rollover just fine.
